sympy/stats/crv.py

```diff
@@ -301,8 +301,7 @@ def compute_quantile(self, **kwargs):
 
         pdf = self.pdf(x)
         cdf = integrate(pdf, (x, left_bound, x), **kwargs)
-
-        quantile = solveset(cdf - p, x, S.Reals)
+        quantile = solveset(cdf - p, x, self.set)
         return Lambda(p, Piecewise((quantile, (p >= 0) & (p <= 1) ), (nan, True)))
```

Comment:
L305 is still returning the **set** `quantile` instead of its (hopefully) single element. The failsafe check should be added here.